{{cookiecutter.repository_name}}/.github/workflows/tests.yml

```diff
@@ -0,0 +1,67 @@
+name: Run Tests
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: {{ '${{ matrix.os }}' }}
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.5, 3.6, 3.7, 3.8]
+        os: [ubuntu-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python {{ '${{ matrix.python-version }}' }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: {{ '${{ matrix.python-version }}' }}
+        
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox tox-gh-actions
+
+    - name: Test with tox
+      run: tox
+
+{%- if cookiecutter.use_codecov_with_ci == 'y' %}
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        file: ./coverage.xml
+{%- endif %}
+
+{%- if cookiecutter.use_pypi_with_ci == 'y' %}
```

Comment:
It seems like this would be executed every time a push is made and only the last step (the actual deploy) will be skipped if there has been no tag.
Can we put this on a separated file which gets triggered only when a tag is created?

Reply:
done, see https://github.com/k15z/python-boilerplate for what this looks like when `use_pypi_with_ci` is true and you create a release